src/init-app.ts

```diff
@@ -100,12 +109,13 @@ export async function initApp() {
   return datasetteWorker;
 }
 
+const BASE_URL = window.location.origin;
 // Intercept events coming from the datasette HTML page.
 function attachEventListeners(output: HTMLElement, datasetteWorker: Worker) {
   function loadPath(path) {
-    path = path.split("#")[0].replace("http://localhost", "");
+    path = path.split("#")[0].replace(BASE_URL, "");
     console.log("Navigating to", { path });
-    history.pushState({ path: path }, path, "#" + path);
+    history.pushState({ path: path }, '', "#" + path);
```

Comment:
Second param is deprecated 

https://developer.mozilla.org/en-US/docs/Web/API/History/pushState